### PR TITLE
Restore walk icon to short trips

### PR DESCRIPTION
**Pull request by amy-corson-ibigroup**
**Description:**

Fixes a small issue that was keeping the walk icon from showing up on shorter trips in itinerary results

**PR Checklist:**
- [x] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [x] Are all languages supported (Internationalization/Localization)?
- [x] Are appropriate Typescript types implemented?

| Before | After |
|--------|-------|
|image|image|

### CODE REVIEW

File: lib/components/narrative/metro/metro-itinerary.tsx [hunk 1]
```diff
@@ -309,7 +305,6 @@ class MetroItinerary extends NarrativeItinerary {
     )
 
     const firstTransitStop = getFirstTransitLegStop(itinerary)
-    const routeLegs = itinerary.legs.filter(removeInsignifigantWalkLegs)
```
Comment (miles-grant-ibigroup) [inline lib/components/narrative/metro/metro-itinerary.tsx:312]: I'm nervous about removing this for all route blocks -- this is used to remove a walking leg icon from a say 3 minute walking, 20 minute transit, 2 minute walking itinerary

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Spotted one potential regression! Could be fixed by probably making the insignificant walk leg removal dependent on the number of legs in the itinerary

File: lib/components/narrative/metro/metro-itinerary.tsx (hunk 1 above)
Comment (amy-corson-ibigroup) [reply lib/components/narrative/metro/metro-itinerary.tsx:312]: This was the only place in the file it was used, which is why I took it out, but I think `removeInsignifigantWalkLegs` is used other places for what you're talking about!

Comment (miles-grant-ibigroup) [reply lib/components/narrative/metro/metro-itinerary.tsx:312]: Yes unfortunately this change affects Walk + Transit itineraries as well. 3 minute walk legs are being shown when they shouldn't be

**Review by miles-grant-ibigroup — APPROVED** [review]
Looks like this may have been a refactor gone wrong. Let's merge this and fix any fallout later

File: lib/components/narrative/metro/metro-itinerary.tsx (hunk 1 above)
Comment (daniel-heppner-ibigroup) [reply lib/components/narrative/metro/metro-itinerary.tsx:312]: Can we do this instead, to address Miles' comment (which I agree with)
```
const routeLegs = itinerary.legs.length > 1 ? itinerary.legs.filter(removeInsignifigantWalkLegs) : itinerary.legs
```

**Review by daniel-heppner-ibigroup — CHANGES_REQUESTED** [review]

File: lib/components/narrative/metro/metro-itinerary.tsx (hunk 1 above)
Comment (amy-corson-ibigroup) [reply lib/components/narrative/metro/metro-itinerary.tsx:312]: Just want to clarify that the routeLegs in this file does not affect transit itineraries - Miles & I discussed this offline and I think that's making this thread confusing (for me at least!!) so apologies for bad transparency!

**Pull request merged**